src/protocol.rs

```diff
@@ -48,6 +48,28 @@ pub enum ProtocolType {
 /// takes as input a [`Message`] and outputs a tuple containing the
 /// participant's output alongside a list of messages to process.
 ///
+/// # 🔒 Message handling
+/// The calling application is responsible for all message sending and routing
+/// for its [`Participant`]. This includes:
+/// 1. Initializing a session by calling [`Participant::initialize_message()`].
+///     The message returned from that function must be passed back to the
+/// [`Participant`] in order to begin the protocol execution.
+/// 2. Passing all messages for the session received from other participants
+/// by calling [`Participant::process_single_message()`].
+/// 3. Routing all messages generated by
+/// [`Participant::process_single_message()`] to the correct recipient.
+///
+/// In order to fulfill these requirements, the calling application must
+/// maintain a mapping from [`ParticipantIdentifier`]s to the appropriate
+/// routing information for that entity.
```

Comment:
I don't think it is entirely obvious how maintaining a mapping fulfills these requirements.

```suggestion
/// [`Message`]s contain a `to: ParticipantIdentifier` field which specifies the recipient of a message. The calling application is responsible for maintaining a mapping from [`ParticipantIdentifier`]s to some network metadata so messages can be routed to the correct `Participant`.
```